pkg/basetypes.go

```diff
@@ -120,6 +120,29 @@ func (p UpkeepPayload) GenerateID() string {
 	return hex.EncodeToString(idh[:])
 }
 
+func (p UpkeepPayload) TriggerID() (string, error) {
+	var extension map[string]interface{}
+	var ok bool
+
+	if extension, ok = p.Trigger.Extension.(map[string]interface{}); !ok {
+		return "", fmt.Errorf("Failed to generate trigger ID for upkeep. Extension is not a map")
+	}
```

Comment:
The extension can be nil so testing for this might be too aggressive of a validity test.

Reply:
I see. What is you recommended behavior when Extension is nil?